### ISSUE TITLE
I need help clearing ram so I can continue my pokemon game

### ISSUE THREAD

**Issue by Firestorm1051** [calc84maniac/tiboyce#71]
Currently if I try to open up pokemon red, it says I need 13857 more bytes of free ram. I don't know how I can get that, and I have clicked no whenever a garbage collect message pops up because I thought it might delete my current save file. Is there a way to get the garbage collect message back so I could free up ram

**Comment by calc84maniac** [comment]
You should either archive or delete programs/variables in RAM you aren't using. Press Enter next to the variable in the memory menu to archive it, or Del to delete it.

**Comment by Firestorm1051** [comment]
Yes, but i have practically archived everything. The only program I have is the tiboyce

**Comment by calc84maniac** [comment]
Oh okay, maybe the issue is that you haven't been Garbage Collecting, which is basically just defragmenting the Archive. If you refuse to Garbage Collect then your save file will be left in RAM instead of being archived, so try making sure your save files are archived as well. Those might be what's taking up the space.

**Comment by Firestorm1051** [comment]
Yes I have done that using 2nd + then Men Management/Delete then App Vars and then I archived everything there and it still says I need 1844 bytes. I have 56557 bytes of ram free at the moment

**Comment by calc84maniac** [comment]
Well, your description of the problem does make me concerned that there might be a memory leak that happens in some cases. Is TI-Boy the only program you've been running on your calculator? And have you been running it through a shell, or just from the home screen?

**Comment by ndye** [comment]
Yeah, that is very odd because i can run Pokemon blue with significantly less ram then described. Either there is a memory leak or another program is causing issues.

**Comment by Firestorm1051** [comment]
Tiboyce is the only program I have and I run it by clicking the program button, selecting tiboyce, and then enter. Once I archived everything then cleared all ram for some reason pokemon red is working again perfectly and I haven’t gained any ram at all from clearing... It just worked

**Comment by calc84maniac** [comment]
I would expect that clearing RAM would have fixed the problem, but I would also expect there to be around 150KB of RAM free after doing so. I'm not totally confident the problem won't happen again. Could you check which OS version you're running in Memory->About? (I doubt this is due to an OS change, but you never know with TI.)

**Comment by Firestorm1051** [comment]
I believe it is 5.3.1.0058, this is right under the ti 84 plus ce and I think that’s the version, the other options are the prod and Id. I got 152k ram back after clearing ram so I think it’s okay

**Comment by ndye** [comment]
Well that's the latest version so that's not it.

**Comment by calc84maniac** [comment]
Oh, I was confused when you said you hadn't gained any RAM from clearing it. Since you only had 56k free after archiving everything before, right?

**Comment by adriweb** [comment]
Just another question although it's probably a moot point now: what's the datecode/hw rev ?
On the back of the calc, to the right of the serial number, there will be something like "L-1016F".

**Comment by Firestorm1051** [comment]
Before I ram cleared, I archived pretty much everything and only had around 50k ram, after I cleared though I randomly got back to 150k although before I archived literally everything before I ram cleared, that’s where my issues started but now it seems to be alright

**Comment by calc84maniac** [comment]
Okay, that's what I was thinking. So I guess I'll have to go over my code again at some point to make sure there aren't any code paths that'll cause a memory leak.
